### PR TITLE
Feat: move dep type to label

### PR DESCRIPTION
**Pull request by lwywoo**
- [x] Tests written and linted
- [ ] Documentation written / README.md updated [https://snyk.io/docs/snyk-for-dotnet/](i)
- [x] Follows [CONTRIBUTING agreement](CONTRIBUTING.md)
- [x] Commit history is tidy [https://git-scm.com/book/en/v2/Git-Branching-Rebasing](i)
- [ ] Reviewed by Snyk team

### What this does
Move property `depType` to be nested in `label` property to be used in depGraph conversion.

### More information
- https://snyksec.atlassian.net/browse/BST-753

### CODE REVIEW

File: test/lib/package-lock.test.ts [hunk 1]
```diff
@@ -116,6 +116,18 @@ test('Parse npm package-lock.json with dev deps only', async (t) => {
...
+test('Parse npm simple package-lock.json dev and prod deps', async (t) => {
+  const expectedDepTree = load('simple-test/expected-tree.json');
+  const depTree = await buildDepTreeFromFiles(
+    `${__dirname}/fixtures/simple-test/`,
+    'package.json',
+    'package-lock.json',
+    true,
+  );
+
+  t.deepEqual(depTree, expectedDepTree, 'Tree is created with dev deps only');
```
Comment (lili2311) [inline test/lib/package-lock.test.ts]: `dev and prod deps` so should the message here also reflect that it is expected to have both?

File: lib/parsers/index.ts [hunk 2]
```diff
@@ -34,7 +34,9 @@ export interface PkgTree {
   meta?: {
     nodeVersion: string;
   };
-  depType?: DepType;
+  labels?: {
+    depType: DepType;
+  };
   hasDevDependencies?: boolean;
   cyclic?: boolean;
   missingLockFileEntry?: boolean;
```
Comment (lili2311) [inline lib/parsers/index.ts]: are these already preserved btw? 
- `cyclic`
- `missingLockFileEntry`
- `cyclic`

File: test/lib/package-lock.test.ts (hunk 1 above)
Comment (lwywoo) [reply test/lib/package-lock.test.ts]: Updated

File: lib/parsers/index.ts (hunk 2 above)
Comment (lwywoo) [reply lib/parsers/index.ts]: - Cycles are not yet supported in depGraph: https://github.com/snyk/dep-graph/blob/master/src/legacy/index.ts#L221-L224
- `missingLockFileEntry` - don't think this is preserved either

**Pull request merged**